### ISSUE TITLE
Polldaddy shortcode callback triggered when shortcode not present

### ISSUE THREAD

**Issue by emmalewis** [Automattic/amp-wp#1620]
When using the plugin in conjunction with the Polldaddy plugin: https://wordpress.org/plugins/polldaddy/ the polldaddy shortcode callback function in `wpcom/class-amp-polldaddy-embed.php` is entered even when there is no polldaddy shortcode on the page. This results in a link at the bottom of the post content with text equal to the post title and url set to `https://polldaddy.com`.

**Comment by westonruter** [comment]
@emmalewis what do you have in your post content to cause this problem? I can't replicate the problem.

**Comment by emmalewis** [comment]
@westonruter Nothing. We have the polldaddy plugin activated but the problem is exclusively in posts that *don't* have a poll in them.

**Comment by emmalewis** [comment]
@westonruter Thanks so much for investigating. We are also on v0.7.2, you can see an example of the link being output at the bottom of this post, immediately underneath the main content. https://scienceblog.cancerresearchuk.org/2017/02/10/3-of-the-toughest-questions-in-cancer-and-more-than-70-million-to-solve-them/amp/

**Comment by westonruter** [comment]
In the non-AMP version of your post, there is a Polldaddy rating embed:

![image](https://user-images.githubusercontent.com/134745/48641128-9e485080-e98d-11e8-98c0-71f7ea1f215a.png)

This is output to the page as:

```html
<div class="pd-rating" id="pd_rating_holder_4922776_post_15110"></div>
```

So that is where the link is coming from. 

You probably have your Rating settings as follows:

![image](https://user-images.githubusercontent.com/134745/48641406-845b3d80-e98e-11e8-836f-d0f979ed704b.png)

So the quick fix is to disable ratings.

**Comment by westonruter** [comment]
A permanent fix can be found in #1621 which should be included as part of 1.0.

**Comment by emmalewis** [comment]
Great, thank you.
